pkg/operator/controller/ingress/controller.go

```diff
@@ -100,6 +100,16 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Index IngressControllers over the default certificate name so that
+	// secretToIngressController can look up IngressControllers that
+	// reference the secret.
+	if err := operatorCache.IndexField(context.Background(), &operatorv1.IngressController{}, "ingressDefaultCertificateName", func(o client.Object) []string {
+		secret := operatorcontroller.RouterEffectiveDefaultCertificateSecretName(o.(*operatorv1.IngressController), operatorcontroller.DefaultOperandNamespace)
```

Comment:
It would be good practice to use `o.GetNamespace()` just in case we ever support ingresscontrollers in other namespaces:
```suggestion
		secret := operatorcontroller.RouterEffectiveDefaultCertificateSecretName(o.(*operatorv1.IngressController), o.GetNamespace())
```
But it isn't really necessary for the release-4.15 branch.

Reply:
Done.